src/main/java/com/j256/ormlite/dao/BaseDaoImpl.java

```diff
@@ -385,6 +386,21 @@ public synchronized T createIfNotExists(T data) throws SQLException {
 		}
 	}
 
+	@Override
+	public synchronized T createIfNotExists(final ID key, final Supplier<T> entityCreator) throws SQLException {
```

Comment:
Sorry to be a pain.  If we are leaving it as Supplier then how about entitySupplier instead of creator?